### DISCUSSION TITLE
Visual indicator for fields that take map keys instead of velocity tokens

### DISCUSSION THREAD

**Discussion by tonygermano** [nextgenhealthcare/connect#4535] [Feature/Enhancement Ideas]
There are a few fields in the GUI that expect to receive the map key name rather than the key's associated value as would be provided by a velocity replacement. It would be helpful to have some visual indicator to alert the user of the proper usage.

**Comment by jonbartels** [comment]
What are some example fields that show each behavior?

I recall you helping a user in Slack this week with one of the FTP writer fields.

I also recall this being specifically discussed in the user guide for the new HTTP header map feature in MC 3.9.x. (Page 354 at https://www.nextgen.com/-/media/files/nextgen-connect/nextgen-connect-39-user-guide.pdf )

**Reply by tonygermano** [reply]
Some other places that come to mind are custom metadata fields on the channel summary tab, thread assignment variables in destination advanced queue settings, and message metadata variables in channel writers.

**Reply by leovander** [reply]
HTTP Receiver Response code is another spot that is unclear/could use a note/indicator on which type of variable is accepted. More of a gripe on that field is having to figure out yourself that your status code which is a number needs to be passed in as a string.

**Comment by tonygermano** [comment]
Related issue https://github.com/nextgenhealthcare/connect/issues/4468
